nixos/modules/tasks/kbd.nix

```diff
@@ -51,13 +57,17 @@ in
     (mkIf setVconsole {
       environment.systemPackages = [ pkgs.kbd ];
 
-      # Let systemd-vconsole-setup.service do the work of setting up the
-      # virtual consoles.  FIXME: trigger a restart of
-      # systemd-vconsole-setup.service if /etc/vconsole.conf changes.
-      environment.etc = [ {
-        target = "vconsole.conf";
-        source = vconsoleConf;
-      } ];
+      environment.etc = [
+        # Let systemd-vconsole-setup.service do the work of setting up the
+        # virtual consoles.
+        { target = "vconsole.conf";
+          source = vconsoleConf;
+        }
+        # Provide kbd with additional packages.
+        { target = "kbd";
+          source = "${kbdEnv}/share";
+        }
+      ];
```

Comment:
Please use `environment.etc."vconsole.conf".source = vconsoleConf`. The list syntax is basically obsolete.